scripts/add.tokens.erc20.mjs

```diff
@@ -0,0 +1,262 @@
+
+export const ${tokenDetails.symbol}_TOKEN_ID: unique symbol = Symbol(${tokenDetails.symbol}_SYMBOL);
+
+export const ${mainnetToken}: RequiredErc20Token = {
+	id: ${tokenDetails.symbol}_TOKEN_ID,
+	network: ETHEREUM_NETWORK,
+	standard: 'erc20',
+	category: 'default',
+	name: '${tokenDetails.name}',
+	symbol: '${tokenDetails.symbol}',
+	decimals: ${tokenDetails.symbol}_DECIMALS,
+	icon: ${tokenName.toLowerCase()},
+	address: '${contractAddress}',
+	exchange: 'erc20',
+	twinTokenSymbol: 'ck${tokenDetails.symbol}'
+};
+
+export const SEPOLIA_${tokenDetails.symbol}_SYMBOL = 'Sepolia${tokenDetails.symbol}';
+
+export const SEPOLIA_${tokenDetails.symbol}_TOKEN_ID: unique symbol = Symbol(SEPOLIA_${tokenDetails.symbol}_SYMBOL);
+
+export const ${testnetToken}: RequiredErc20Token = {
+	id: SEPOLIA_${tokenDetails.symbol}_TOKEN_ID,
+	network: SEPOLIA_NETWORK,
+	standard: 'erc20',
+	category: 'default',
+	name: '${tokenDetails.name}',
+	symbol: '${tokenDetails.symbol}',
+	decimals: ${tokenDetails.symbol}_DECIMALS,
+	icon: ${tokenName.toLowerCase()},
+	address: '${testnetContractAddress}',
+	exchange: 'erc20',
+	twinTokenSymbol: 'ckSepolia${tokenDetails.symbol}'
+};
+`;
+
+	writeFileSync(newFilePath, newContent);
+	console.log(`Created ${newFilePath}`);
+
+	return { newFileName, mainnetToken, testnetToken };
+};
+
+const updateTokensErc20Env = (newFileName, mainnetToken, testnetToken) => {
+	const filePath = path.join(DATA_DIR_PATH, 'tokens.erc20.env.ts');
+	let content = readFileSync(filePath, 'utf8');
+
+	const regexList = [
+		{
+			regex: /const ERC20_TWIN_TOKENS_MAINNET: RequiredErc20Token\[] = \[([\s\S]*?)];/,
+			token: mainnetToken
+		},
+		{
+			regex: /const ERC20_TWIN_TOKENS_SEPOLIA: RequiredErc20Token\[] = \[([\s\S]*?)];/,
+			token: testnetToken
+		}
+	];
+
+	regexList.forEach(({ regex, token }) => {
+		const match = content.match(regex);
+		if (match) {
+			const tokensList = match[1].trim();
+			const updatedTokensList = tokensList ? `${tokensList}, ${token}` : token;
+			content = content.replace(
+				regex,
+				`${match[0].split(':')[0]}: RequiredErc20Token[] = [${updatedTokensList}];`
+			);
+		}
+	});
+
+	const importStatement = `import { ${mainnetToken}, ${testnetToken} } from '$env/${newFileName.replace('.ts', '')}';\n`;
+	content = importStatement + content;
+
+	writeFileSync(filePath, content);
+	console.log(`Updated ${filePath}`);
+};
+
+const flattenData = (data) => {
+	return Object.keys(data).map(symbol => ({
+		symbol,
+		...data[symbol]
+	}));
+};
+
+const flattenEnvironmentData = (data) => {
+	return Object.entries(data).reduce((acc, [environment, values]) => ({
+		...acc,
+		[environment]: flattenData(values)
+	}), {});
+};
+
+const loadSupportedTokens = () => {
+	const jsonPath = path.resolve(DATA_DIR_PATH, 'tokens.ckerc20.json');
+	return flattenEnvironmentData(JSON.parse(readFileSync(jsonPath, 'utf-8')));
+};
+
+const filterTokens = async (prodTokens, testnetTokens) => {
```

Comment:
Does this approach actually works when the tokens exists on testnet but, has not yet been proposed and executed on mainnet?

For example ckSepoliaPEPE is on testnet but, the proposal is not yet on mainnet.

Reply:
no, for now it works only for tokens that are both testnet and mainnet, i thought as a first step to do it like that, then we could improve

what do you think?

Reply:
Ah, I think we should change it then 😉. The typical flow is deploy a ckSepolia token on mainnet for testnet / test purpose and then few days/weeks later, make a NNS proposal to enable the related production token. So, we want to integrate testnet earlier as this week with ckSepoliaPepe without ckPepe yet. Does that make sense?

Reply:
ah! yes, for sure it makes sense, let me modify it and come back

Reply:
updated! but it became another script basically eheh! please review it all, thank you!